packages/bridging/src/providers/near-intents/NearIntentsBridgeProvider.ts

```diff
@@ -0,0 +1,370 @@
+import {
+  NEAR_INTENTS_BLOCKCHAIN_TO_NATIVE_WRAPPED_TOKEN_ADDRESS,
+  NEAR_INTENTS_STATUS_TO_COW_STATUS,
+  NEAR_INTENTS_SUPPORTED_NETWORKS,
+} from './const'
+import { adaptToken, adaptTokens, calculateDeadline, getTokenByAddressAndChainId } from './util'
+
+import type { cowAppDataLatestScheme as latestAppData } from '@cowprotocol/sdk-app-data'
+import type { AbstractProviderAdapter, SignerLike } from '@cowprotocol/sdk-common'
+import type { ChainId, ChainInfo, EvmCall, SupportedChainId, TokenInfo } from '@cowprotocol/sdk-config'
+import type { CowShedSdkOptions } from '@cowprotocol/sdk-cow-shed'
+import type { Hex } from 'viem'
+import type {
+  BridgeDeposit,
+  BridgeHook,
+  BridgeProvider,
+  BridgeProviderInfo,
+  BridgeQuoteResult,
+  BridgeStatusResult,
+  BridgingDepositParams,
+  BuyTokensParams,
+  GetProviderBuyTokens,
+  QuoteBridgeRequest,
+} from '../../types'
+
+export interface NearIntentsQuoteResult extends BridgeQuoteResult {
+  depositAddress: Hex
+}
+
+export const NEAR_INTENTS_HOOK_DAPP_ID = `${HOOK_DAPP_BRIDGE_PROVIDER_PREFIX}/near-intents`
+
+export interface NearIntentsBridgeProviderOptions {
+  cowShedOptions?: CowShedSdkOptions
+}
+
+export class NearIntentsBridgeProvider implements BridgeProvider<NearIntentsQuoteResult> {
+  protected api: NearIntentsApi
+  protected cowShedSdk: CowShedSdk
+
+  info: BridgeProviderInfo = {
+    name: 'Near Intents',
+    logoUrl: `${RAW_PROVIDERS_FILES_PATH}/near-intents/near-intents-logo.png`,
+    dappId: NEAR_INTENTS_HOOK_DAPP_ID,
+    website: 'https://www.near.org/intents',
+  }
+
+  constructor(options?: NearIntentsBridgeProviderOptions, _adapter?: AbstractProviderAdapter) {
+    const adapter = _adapter || options?.cowShedOptions?.adapter
+    if (adapter) {
+      setGlobalAdapter(adapter)
+    }
+    this.api = new NearIntentsApi()
+    this.cowShedSdk = new CowShedSdk(adapter, options?.cowShedOptions?.factoryOptions)
+  }
+
+  async getNetworks(): Promise<ChainInfo[]> {
+    return NEAR_INTENTS_SUPPORTED_NETWORKS
+  }
+
+  async getBuyTokens(params: BuyTokensParams): Promise<GetProviderBuyTokens> {
+    const tokens = adaptTokens(await this.api.getTokens())
+    return {
+      tokens: tokens.filter((token) => token.chainId === params.buyChainId),
+      isRouteAvailable: tokens.length > 0,
+    }
+  }
+
+  async getIntermediateTokens(request: QuoteBridgeRequest): Promise<TokenInfo[]> {
+    if (request.kind !== OrderKind.SELL) {
+      throw new BridgeProviderQuoteError(BridgeQuoteErrors.ONLY_SELL_ORDER_SUPPORTED, { kind: request.kind })
+    }
+
+    const { sellTokenChainId, buyTokenChainId, buyTokenAddress } = request
+
+    const tokens = adaptTokens(await this.api.getTokens())
+    const sourceTokens = tokens.filter((token) => token.chainId === sellTokenChainId)
+    const targetTokens = tokens.filter((token) => token.chainId === buyTokenChainId)
+
+    const wrappedTokenAddresses = Object.values(NEAR_INTENTS_BLOCKCHAIN_TO_NATIVE_WRAPPED_TOKEN_ADDRESS).map((a) =>
```

Comment:
It feels like overkill way to do the check later if its wrapped token, it could very well use `WRAPPED_NATIVE_CURRENCIES` alternativelly, we could make a common util `isWrappedNativeCurrency(chainId, tokenAddress)`